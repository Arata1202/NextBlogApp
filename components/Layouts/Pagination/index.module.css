.container {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 24px;
  margin-top: 24px;
}
.list {
  margin: 0 4px;
}
.item {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 36px;
  height: 36px;
  border-radius: var(--border-radius);
}
.item:hover {
  color: rgb(59 130 246);
}
.current {
  background-color: var(--color-current);
}
