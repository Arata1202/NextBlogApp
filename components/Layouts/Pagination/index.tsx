'use client';
import { useTheme } from 'next-themes';
import React, { useMemo } from 'react';
import styles from './index.module.css';
import { LIMIT } from '@/constants';

type Props = {
  totalCount: number;
  current?: number;
  basePath?: string;
  q?: string;
};

const Pagination: React.FC<Props> = React.memo(({ totalCount, current = 1, basePath = '', q }) => {
  const pages = useMemo(() => {
    return Array.from({ length: Math.ceil(totalCount / LIMIT) }, (_, i) => i + 1);
  }, [totalCount]);

  const getPageLink = (page: number) => {
    return `${basePath}/p/${page}${q ? `?q=${q}` : ''}`;
  };

  const { theme } = useTheme();

  return (
    <ul className={`${styles.container} PaginationContainer`}>
      {pages.map((p) => (
        <li className={styles.list} key={p}>
          {current !== p ? (
            <a href={getPageLink(p)}>
              <p
                className={`hover:text-blue-500 ${styles.item} ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
              >
                {p}
              </p>
            </a>
          ) : (
            <span
              className={`hover:text-blue-500 ${styles.item} ${theme === 'dark' ? 'bg-gray-500 text-white' : 'bg-gray-300 text-gray-700'}`}
              aria-current="page"
            >
              {p}
            </span>
          )}
        </li>
      ))}
    </ul>
  );
});

Pagination.displayName = 'Pagination';

export default Pagination;
