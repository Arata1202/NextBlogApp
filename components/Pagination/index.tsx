//最適化済み

import Link from 'next/link';
import React, { useMemo } from 'react';
import styles from './index.module.css';
import { LIMIT } from '@/constants';

type Props = {
  totalCount: number;
  current?: number;
  basePath?: string;
  q?: string;
};

const Pagination: React.FC<Props> = React.memo(({ totalCount, current = 1, basePath = '', q }) => {
  const pages = useMemo(() => {
    return Array.from({ length: Math.ceil(totalCount / LIMIT) }, (_, i) => i + 1);
  }, [totalCount]);

  const getPageLink = (page: number) => {
    return `${basePath}/p/${page}${q ? `?q=${q}` : ''}`;
  };

  return (
    <ul className={styles.container}>
      {pages.map((p) => (
        <li className={styles.list} key={p}>
          {current !== p ? (
            <Link href={getPageLink(p)}>
              <a className={styles.item}>{p}</a>
            </Link>
          ) : (
            <span className={`${styles.item} ${styles.current}`} aria-current="page">
              {p}
            </span>
          )}
        </li>
      ))}
    </ul>
  );
});

Pagination.displayName = 'Pagination';

export default Pagination;
