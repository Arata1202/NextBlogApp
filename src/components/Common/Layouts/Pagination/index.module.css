.container {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 24px;
  margin-top: 24px;
}
.list {
  margin: 0 4px;
}
.item {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 36px;
  height: 36px;
  border-radius: var(--border-radius);
}

@media (max-width: 768px) {
  .container {
    margin-top: 0;
  }
}

@media (max-width: 640px) {
  .container {
    margin-bottom: 1.75rem;
  }
}
