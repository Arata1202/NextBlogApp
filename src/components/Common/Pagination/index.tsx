'use client';

import Link from 'next/link';
import { useTheme } from 'next-themes';
import { LIMIT } from '@/constants/limit';
import styles from './index.module.css';

type Props = {
  totalCount: number;
  current?: number;
  basePath?: string;
  q?: string;
};

export default function Pagination({ totalCount, current = 1, basePath = '', q }: Props) {
  const { theme } = useTheme();

  const pages = Array.from({ length: Math.ceil(totalCount / LIMIT) }).map((_, i) => i + 1);

  return (
    <>
      <ul className={styles.container}>
        {pages.map((p) => (
          <li className={styles.list} key={p}>
            {current !== p ? (
              <Link
                href={`${basePath}/p/${p}` + (q ? `?q=${q}` : '')}
                className={`${styles.item} hover:text-blue-500 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
              >
                {p}
              </Link>
            ) : (
              <span
                className={`${styles.item} ${theme === 'dark' ? 'bg-gray-500 text-white' : 'bg-gray-300 text-gray-700'}`}
              >
                {p}
              </span>
            )}
          </li>
        ))}
      </ul>
    </>
  );
}
