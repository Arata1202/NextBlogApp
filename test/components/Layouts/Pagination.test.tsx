import { render } from '@testing-library/react';
import Pagination from '@/components/Layouts/Pagination';

describe('Pagination', () => {
  test('スナップショット（Pagination）', () => {
    const { asFragment } = render(<Pagination totalCount={100} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
