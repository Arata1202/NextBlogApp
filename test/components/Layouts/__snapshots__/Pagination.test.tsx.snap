// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination スナップショット（Pagination） 1`] = `
<DocumentFragment>
  <ul
    class="container PaginationContainer"
  >
    <li
      class="list"
    >
      <span
        aria-current="page"
        class="item bg-gray-300 text-gray-700"
      >
        1
      </span>
    </li>
    <li
      class="list"
    >
      <a
        href="/p/2"
      >
        <p
          class="hover:text-blue-500 item LightTheme"
        >
          2
        </p>
      </a>
    </li>
    <li
      class="list"
    >
      <a
        href="/p/3"
      >
        <p
          class="hover:text-blue-500 item LightTheme"
        >
          3
        </p>
      </a>
    </li>
    <li
      class="list"
    >
      <a
        href="/p/4"
      >
        <p
          class="hover:text-blue-500 item LightTheme"
        >
          4
        </p>
      </a>
    </li>
    <li
      class="list"
    >
      <a
        href="/p/5"
      >
        <p
          class="hover:text-blue-500 item LightTheme"
        >
          5
        </p>
      </a>
    </li>
    <li
      class="list"
    >
      <a
        href="/p/6"
      >
        <p
          class="hover:text-blue-500 item LightTheme"
        >
          6
        </p>
      </a>
    </li>
    <li
      class="list"
    >
      <a
        href="/p/7"
      >
        <p
          class="hover:text-blue-500 item LightTheme"
        >
          7
        </p>
      </a>
    </li>
    <li
      class="list"
    >
      <a
        href="/p/8"
      >
        <p
          class="hover:text-blue-500 item LightTheme"
        >
          8
        </p>
      </a>
    </li>
    <li
      class="list"
    >
      <a
        href="/p/9"
      >
        <p
          class="hover:text-blue-500 item LightTheme"
        >
          9
        </p>
      </a>
    </li>
    <li
      class="list"
    >
      <a
        href="/p/10"
      >
        <p
          class="hover:text-blue-500 item LightTheme"
        >
          10
        </p>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
